packages/next/server/base-server.ts

```diff
@@ -1275,9 +1278,13 @@ export default abstract class Server {
       resolvedUrlPathname = stripNextDataPath(resolvedUrlPathname)
       urlPathname = stripNextDataPath(urlPathname)
     }
+    const canLeveragePreviousCache = !!req.headers['x-nextjs-dedupe']
```

Comment:
let's nuke this, the fact that we validate the cache entry isn't expired before returning it is sufficient.